db.js

```diff
@@ -164,10 +164,109 @@ var friends = (function () {
   };
 })();
 
+var libraries = (function () {
+  return {
+    getUsersLibrary: async function (userId) {
+      var userQuery = "SELECT * FROM users WHERE id = ?";
+      var inserts = [userId];
+      userQuery = mysql.format(userQuery, inserts);
+
+      // TODO: Add in categories field to book query.
+      var bookQuery = "SELECT * FROM books WHERE user_id = ?";
+      var inserts = [userId];
+      bookQuery = mysql.format(bookQuery, inserts);
+
+      let user = await pool.query(userQuery).then(([rows, fields]) => {
+        return rows;
+      });
+
+      let books = await pool.query(bookQuery).then(([rows, fields]) => {
+        return rows;
+      });
+
+      for (var i = 0; i < books.length; i++) {
+        var data = books[i].id;
+
+        var loanQuery = `SELECT * FROM loans WHERE book_id = '${data}' ` +
+                        `ORDER BY accept_date DESC LIMIT 1`;
+
+        var loans = await pool.query(loanQuery).then(([rows, fields]) => {
+          if (!rows || rows.length === 0) {
+            return null;
+          }
+          return rows[0];
+        });
+
+        var loanCountQuery = `SELECT COUNT (*) as "count" FROM loans WHERE book_id = '${data}'`;
+
+        var loanCount = await pool
+          .query(loanCountQuery)
+          .then(([rows, fields]) => {
+            if (!rows || rows.length === 0) {
+              return null;
+            }
+            return rows[0].count;
+          });
+
+        books[i].loan_count = loanCount;
+        books[i].loan = loans;
+      }
+      return { user, books };
+    },
+
+    getLibraryById: function (userId) {
+      var userQuery = "SELECT * FROM users WHERE id = ?";
```

Comment:
You also want to get the friend status here. You'll have to pass in both the current user's id and the target user's id into the function and then join the user and friendship tables. (I also noticed that the GET friends route is broken, but that's for another day)